contracts/swaps/SwapsUser.sol

```diff
@@ -124,24 +127,45 @@ contract SwapsUser is State, SwapsEvents, FeesHelper {
         } else {
             require(vals[0] <= vals[1], "min greater than max");
         }
-
-        require(loanDataBytes.length == 0, "invalid state");
-        (destTokenAmountReceived, sourceTokenAmountUsed) = _swapsCall_internal(
-            addrs,
-            vals
-        );
-
+        if (loanDataBytes.length == 0) {
```

Comment:
this whole usecase can be optimized
if loanDataBytes.length == 0 && not (flags & DEX_SELECTOR_FLAG != 0) 
normal
else
custom

Reply:
The flags variable is part of loanDataBytes and to get a value for it loanDataBytes has to be decoded which means that if loanDataBytes were to have a length of 0 or not match the formatting expected when being decoded to get flags and bytes[] array of payloads, an error would be thrown as the decode will fail